### PR TITLE
[CAMEL-16554] Add support for private keys with OPENSSH PRIVATE KEY header

### PR DESCRIPTION
**Pull request by grgrzybek**

### CODE REVIEW

File: components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceHelperKeyPairProvider.java [hunk 1]
```diff
@@ -126,6 +128,36 @@ public void setPasswordFinder(Supplier<char[]> passwordFinder) {
             InputStreamReader isr = null;
             InputStream is = null;
             try {
+                is = ResourceHelper.resolveMandatoryResourceAsInputStream(camelContext, resource);
```
Comment (oscerd) [inline components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceHelperKeyPairProvider.java:131]: LGTM to me, I would use a try-with-resource here nevertheless

**Review by oscerd — APPROVED** [review]
LGTM, just a little comment.

File: components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceHelperKeyPairProvider.java (hunk 1 above)
Comment (grgrzybek) [reply components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceHelperKeyPairProvider.java:131]: This is the same line as for "normal" path, where `is` is closed using `IoUtils.closeQuietly(r, is, isr)` in finally block.
`org.apache.sshd.common.util.security.SecurityUtils#loadKeyPairIdentities()` consumes the stream, so there are two such lines.

Comment (oscerd) [reply components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceHelperKeyPairProvider.java:131]: Ah yeah, didn't notice that. All good then. Merge if you want.

Comment (grgrzybek) [reply components/camel-ssh/src/main/java/org/apache/camel/component/ssh/ResourceHelperKeyPairProvider.java:131]: let's wait for CI.

**Comment by grgrzybek** [comment]
The checks have passed. Merging.

**Pull request merged**